src/riak_kv_eleveldb_backend.erl

```diff
@@ -173,19 +175,44 @@ get(Bucket, Key, #state{read_opts=ReadOpts,
             {error, Reason, State}
     end.
 
-%% @doc Insert an object into the eleveldb backend.
 -type index_spec() :: {add, Index, SecondaryKey} | {remove, Index, SecondaryKey}.
+
+%% @doc Normal put, use existing option, do not modify write options
 -spec put(riak_object:bucket(), riak_object:key(), [index_spec()], binary(), state()) ->
                  {ok, state()} |
                  {error, term(), state()}.
-put(Bucket, PrimaryKey, IndexSpecs, Val, #state{ref=Ref,
+put(Bucket, PrimaryKey, IndexSpecs, Val, State) ->
+    Sync = false,
+    do_put(Bucket, PrimaryKey, IndexSpecs, Val, Sync, State).
+
+%% @doc put_flush capability - do a put with a flush to disk
+-spec flush_put(riak_object:bucket(), riak_object:key(), [index_spec()], binary(), state()) ->
+                 {ok, state()} |
+                 {error, term(), state()}.
+flush_put(Bucket, PrimaryKey, IndexSpecs, Val, State) ->
+    Sync = true,
+    do_put(Bucket, PrimaryKey, IndexSpecs, Val, Sync, State).
+
+%% @doc Insert an object into the eleveldb backend.
+-spec do_put(riak_object:bucket(), riak_object:key(), [index_spec()], binary(), boolean(), state()) ->
+                 {ok, state()} |
+                 {error, term(), state()}.
+do_put(Bucket, PrimaryKey, IndexSpecs, Val, Sync, #state{ref=Ref,
                                                 write_opts=WriteOpts,
                                                 legacy_indexes=WriteLegacy,
                                                 fixed_indexes=FixedIndexes}=State) ->
     %% Create the KV update...
     StorageKey = to_object_key(Bucket, PrimaryKey),
     Updates1 = [{put, StorageKey, Val} || Val /= undefined],
 
+    %% Setup write options...
+    WriteOpts2 = case Sync of
+        true ->
+            lists:keyreplace(sync,1,WriteOpts, {sync,Sync});
```

Comment:
This relies on the presence of sync in the eleveldb cuttlefish:

https://github.com/basho/eleveldb/blob/develop-3.0/priv/eleveldb.schema#L32-L39

which then becomes part of the write_opts at startup:

https://github.com/basho/riak_kv/blob/mas-i350-selectivesync/src/riak_kv_eleveldb_backend.erl#L633-L635

So it will always be there, base don the current logic.  However, this does appear to be quite a brittle chain.